pkg/common/peertracker/tracker_linux.go

```diff
@@ -81,6 +102,7 @@ func (l *linuxWatcher) IsAlive() error {
 	defer l.mtx.Unlock()
 
 	if l.procfd < 0 {
+		l.log.Warn("Caller is no longer being watched")
```

Comment:
I'm a little torn here. While in general I think creating structured fields is a good idea, I can't help but notice that this logging could also be accomplished with a simple refactor:

```
func (l *linuxWatcher) IsAlive() error {
	if err := l.isAlive(); err != nil {
		l.log.WithError(err).Warn("Peertracker liveness check failed")
		return err
	}
	return nil
}

func (l *linuxWatcher) isAlive() error {
	// ... current IsAlive impl ...
}
```

Reply:
True, though we'd lose the searchable old vs. new value fields, which could make it harder to organize or analyze logging data without first needing to do string/regex parsing.

Reply:
Fair enough!